arc/conformance/plugin/arc_conformance.sh

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env bash
+
+set -e
+
+results_dir="${RESULTS_DIR:-/tmp/results}"
```

Comment:
I've opened https://github.com/Azure/secrets-store-csi-driver-provider-azure/pull/783 to enable `shellcheck`. I ran that with the script and there are some errors. Could you run shellcheck after the PR is merged and resolve the issues in the script?

Reply:
It seem to be [passing](https://dev.azure.com/AzureContainerUpstream/Secrets%20Store%20CSI%20Driver%20Provider%20Azure/_build/results?buildId=36416&view=logs&j=3a559e2a-952e-58d2-b8db-2e604a9266d7&t=4d1d8deb-ff81-527f-3e13-6a731fe8f4eb).

Reply:
Update shellcheck to `$(SHELLCHECK) *.sh` locally and run it. `*/*.sh` seems to be acting weirdly. 

```bash
/home/anramase/go/src/github.com/Azure/secrets-store-csi-driver-provider-azure/.tools/bin/shellcheck-v0.7.2 *.sh

In Untitled-1.sh line 15:
    for i in $(seq 1 $max_retries)
    ^-^ SC2034: i appears unused. Verify use (or export if used externally).


In Untitled-1.sh line 17:
    if [[ $(kubectl wait --for=condition=available ${RESOURCE} --all --namespace ${NAMESPACE}) ]]; then
                                                   ^---------^ SC2086: Double quote to prevent globbing and word splitting.
                                                                                 ^----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
    if [[ $(kubectl wait --for=condition=available "${RESOURCE}" --all --namespace "${NAMESPACE}") ]]; then


In Untitled-1.sh line 29:
  cd ${results_dir}
     ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  cd "${results_dir}"


In Untitled-1.sh line 31:
  tar czf results.tar.gz *
                         ^-- SC2035: Use ./*glob* or -- *glob* so names with dashes won't become options.


In Untitled-1.sh line 33:
  printf ${results_dir}/results.tar.gz > ${results_dir}/done
         ^---------------------------^ SC2059: Don't use variables in the printf format string. Use printf '..%s..' "$foo".
         ^------------^ SC2086: Double quote to prevent globbing and word splitting.
                                         ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  printf "${results_dir}"/results.tar.gz > "${results_dir}"/done


In Untitled-1.sh line 51:
  --name $keyvault_resource_group \
         ^----------------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --name "$keyvault_resource_group" \


In Untitled-1.sh line 52:
  --location $keyvault_location 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
             ^----------------^ SC2086: Double quote to prevent globbing and word splitting.
                                   ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --location "$keyvault_location" 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 56:
  --name $keyvault_name \
         ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --name "$keyvault_name" \


In Untitled-1.sh line 57:
  --resource-group $keyvault_resource_group \
                   ^----------------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --resource-group "$keyvault_resource_group" \


In Untitled-1.sh line 58:
  --location $keyvault_location \
             ^----------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --location "$keyvault_location" \


In Untitled-1.sh line 59:
  --sku premium 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                   ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --sku premium 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 64:
  --name $keyvault_name \
         ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --name "$keyvault_name" \


In Untitled-1.sh line 65:
  --resource-group $keyvault_resource_group \
                   ^----------------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --resource-group "$keyvault_resource_group" \


In Untitled-1.sh line 66:
  --spn ${AZURE_CLIENT_ID} \
        ^----------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --spn "${AZURE_CLIENT_ID}" \


In Untitled-1.sh line 69:
  --certificate-permissions get create import 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                                                 ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --certificate-permissions get create import 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 74:
  --vault-name $keyvault_name \
               ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --vault-name "$keyvault_name" \


In Untitled-1.sh line 76:
  --value $secret_value 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
          ^-----------^ SC2086: Double quote to prevent globbing and word splitting.
                           ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --value "$secret_value" 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 84:
  --vault-name $keyvault_name \
               ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --vault-name "$keyvault_name" \


In Untitled-1.sh line 87:
  --size 2048 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                 ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --size 2048 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 90:
  --vault-name $keyvault_name \
               ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --vault-name "$keyvault_name" \


In Untitled-1.sh line 93:
  -f publicKey.pem 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                      ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  -f publicKey.pem 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 95:
  keyVersion=$(az keyvault key show --vault-name $keyvault_name --name $key_name --query "key.kid" | tr -d '"' | sed 's#.*/##')
                                                 ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  keyVersion=$(az keyvault key show --vault-name "$keyvault_name" --name $key_name --query "key.kid" | tr -d '"' | sed 's#.*/##')


In Untitled-1.sh line 97:
  export KEY_VALUE=$(cat publicKey.pem)
         ^-------^ SC2155: Declare and assign separately to avoid masking return values.


In Untitled-1.sh line 102:
  --vault-name $keyvault_name \
               ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --vault-name "$keyvault_name" \


In Untitled-1.sh line 105:
  --size 2048 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                 ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --size 2048 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 109:
  --vault-name $keyvault_name \
               ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --vault-name "$keyvault_name" \


In Untitled-1.sh line 112:
  --curve P-256 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                   ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --curve P-256 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 124:
  --insecure 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --insecure 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 126:
  openssl pkcs12 -export -in test.crt -inkey test.key -out test.pfx -passout pass: 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                                                                                      ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  openssl pkcs12 -export -in test.crt -inkey test.key -out test.pfx -passout pass: 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 129:
  --vault-name $keyvault_name \
               ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --vault-name "$keyvault_name" \


In Untitled-1.sh line 131:
  --file test.pfx 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                     ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --file test.pfx 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 134:
  --vault-name $keyvault_name \
               ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --vault-name "$keyvault_name" \


In Untitled-1.sh line 136:
  --file test.pfx 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                     ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --file test.pfx 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 146:
  --insecure 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --insecure 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 148:
  openssl pkcs12 -export -in testec.crt -inkey testec.key -out testec.pfx -passout pass: 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                                                                                            ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  openssl pkcs12 -export -in testec.crt -inkey testec.key -out testec.pfx -passout pass: 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 151:
  --vault-name $keyvault_name \
               ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --vault-name "$keyvault_name" \


In Untitled-1.sh line 153:
  --file testec.pfx 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                       ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --file testec.pfx 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 158:
  echo "ERROR: parameter TENANT_ID is required." > ${results_dir}/error
                                                   ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  echo "ERROR: parameter TENANT_ID is required." > "${results_dir}"/error


In Untitled-1.sh line 163:
  echo "ERROR: parameter SUBSCRIPTION_ID is required." > ${results_dir}/error
                                                         ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  echo "ERROR: parameter SUBSCRIPTION_ID is required." > "${results_dir}"/error


In Untitled-1.sh line 168:
  echo "ERROR: parameter AZURE_CLIENT_ID is required." > ${results_dir}/error
                                                         ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  echo "ERROR: parameter AZURE_CLIENT_ID is required." > "${results_dir}"/error


In Untitled-1.sh line 173:
  echo "ERROR: parameter AZURE_CLIENT_SECRET is required." > ${results_dir}/error
                                                             ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  echo "ERROR: parameter AZURE_CLIENT_SECRET is required." > "${results_dir}"/error


In Untitled-1.sh line 178:
  echo "ERROR: parameter ARC_CLUSTER_NAME is required." > ${results_dir}/error
                                                          ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  echo "ERROR: parameter ARC_CLUSTER_NAME is required." > "${results_dir}"/error


In Untitled-1.sh line 183:
  echo "ERROR: parameter ARC_CLUSTER_RG is required." > ${results_dir}/error
                                                        ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  echo "ERROR: parameter ARC_CLUSTER_RG is required." > "${results_dir}"/error


In Untitled-1.sh line 193:
  -u ${AZURE_CLIENT_ID} \
     ^----------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  -u "${AZURE_CLIENT_ID}" \


In Untitled-1.sh line 194:
  -p ${AZURE_CLIENT_SECRET} \
     ^--------------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  -p "${AZURE_CLIENT_SECRET}" \


In Untitled-1.sh line 195:
  --tenant ${TENANT_ID} 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
           ^----------^ SC2086: Double quote to prevent globbing and word splitting.
                           ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  --tenant "${TENANT_ID}" 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 197:
az account set --subscription ${SUBSCRIPTION_ID} 2> ${results_dir}/error || python3 /arc/setup_failure_handler.py
                              ^----------------^ SC2086: Double quote to prevent globbing and word splitting.
                                                    ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
az account set --subscription "${SUBSCRIPTION_ID}" 2> "${results_dir}"/error || python3 /arc/setup_failure_handler.py


In Untitled-1.sh line 208:
    echo "ERROR: deployment is not avilable in namespace - azure-arc" > ${results_dir}/error
                                                                        ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
    echo "ERROR: deployment is not avilable in namespace - azure-arc" > "${results_dir}"/error


In Untitled-1.sh line 217:
      --cluster-name ${ARC_CLUSTER_NAME} \
                     ^-----------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
      --cluster-name "${ARC_CLUSTER_NAME}" \


In Untitled-1.sh line 218:
      --resource-group ${ARC_CLUSTER_RG} \
                       ^---------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
      --resource-group "${ARC_CLUSTER_RG}" \


In Untitled-1.sh line 230:
/arc/e2e -ginkgo.v -ginkgo.skip=${GINKGO_SKIP}
                                ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
/arc/e2e -ginkgo.v -ginkgo.skip="${GINKGO_SKIP}"


In Untitled-1.sh line 233:
az k8s-extension delete --name arc-akv-conformance --resource-group ${ARC_CLUSTER_RG} --cluster-type connectedClusters --cluster-name ${ARC_CLUSTER_NAME} --force --yes --no-wait
                                                                    ^---------------^ SC2086: Double quote to prevent globbing and word splitting.
                                                                                                                                      ^-----------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
az k8s-extension delete --name arc-akv-conformance --resource-group "${ARC_CLUSTER_RG}" --cluster-type connectedClusters --cluster-name "${ARC_CLUSTER_NAME}" --force --yes --no-wait


In Untitled-1.sh line 234:
az group delete --name $keyvault_resource_group --yes --no-wait
                       ^----------------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
az group delete --name "$keyvault_resource_group" --yes --no-wait

For more information:
  https://www.shellcheck.net/wiki/SC2034 -- i appears unused. Verify use (or ...
  https://www.shellcheck.net/wiki/SC2155 -- Declare and assign separately to ...
  https://www.shellcheck.net/wiki/SC2035 -- Use ./*glob* or -- *glob* so name...
make: *** [Makefile:84: shellcheck] Error 1
```

Reply:
Should be fixed with https://github.com/Azure/secrets-store-csi-driver-provider-azure/pull/786.